exporter/chronicleexporter/marshal.go

```diff
@@ -38,43 +43,42 @@ var supportedLogTypes = map[string]string{
 	"sql_server":                "MICROSOFT_SQL",
 }
 
-type marshaler struct {
-	cfg          Config
-	labels       []label
-	teleSettings component.TelemetrySettings
-}
-
-type payload struct {
-	Entries    []entry `json:"entries"`
-	CustomerID string  `json:"customer_id"`
-	LogType    string  `json:"log_type"`
-	Namespace  string  `json:"namespace,omitempty"`
-	Labels     []label `json:"labels,omitempty"`
+//go:generate mockery --name logMarshaler --filename mock_log_marshaler.go --structname MockMarshaler --inpackage
+type logMarshaler interface {
+	MarshalRawLogs(ctx context.Context, ld plog.Logs) ([]*generated.BatchCreateLogsRequest, error)
 }
 
-type label struct {
-	Key   string `json:"key"`
-	Value string `json:"value"`
+type protoMarshaler struct {
+	cfg          Config
+	teleSettings component.TelemetrySettings
+	labels       []*generated.Label
+	startTime    time.Time
+	customerID   []byte
+	collectorID  []byte
 }
 
-type entry struct {
-	LogText   string `json:"log_text"`
-	Timestamp string `json:"ts_rfc3339"`
-}
+func newProtoMarshaler(cfg Config, teleSettings component.TelemetrySettings, labels []*generated.Label) (*protoMarshaler, error) {
+	col, err := uuid.Parse(collectorID)
+	if err != nil {
+		return nil, fmt.Errorf("parse collector ID: %w", err)
+	}
 
-type logMarshaler interface {
-	MarshalRawLogs(ctx context.Context, ld plog.Logs) ([]payload, error)
-}
+	CustomerID, err := uuid.Parse(cfg.CustomerID)
+	if err != nil {
+		return nil, fmt.Errorf("parse customer ID: %w", err)
+	}
 
-func newMarshaler(cfg Config, teleSettings component.TelemetrySettings, labels []label) *marshaler {
-	return &marshaler{
+	return &protoMarshaler{
+		startTime:    time.Now(),
```

Comment:
Do we need this to be UTC time or is localized fine?

Reply:
ope already set it up for UTC. probably doesn't hurt